### ISSUE TITLE
Hello. Is there some example of application where you use this package?

### ISSUE THREAD

**Issue by antizeus** [arcamens/django-debits#14]
I need to hook a paypal payment system. I'm very happy to have found this package although the API seems very confusing. It would be much better if there were a demo which works out of the box so people can just test and reuse the code in their application.

**Comment by vporton** [comment]
A rather advanced demo (both one-time and subscription payments) exists: `debits/debits_test/`.

**Comment by antizeus** [comment]
How can i just run that? Can you set it up online so i can test it?

**Comment by vporton** [comment]
> How can i just run that? Can you set it up online so i can test it?

```sh
cd debits/debits_test/
./manage.py makemigrations
./manage.py migrate
./manage.py loaddata ../debits_base/fixtures/processors.json fixtures/pricingplans.json fixtures/products.json
./manage.py runserver
```

**Comment by antizeus** [comment]
I'm sorry but it doesn't make any sense. Can you explain exactly what you intent with that UI? it is really confusing. I have no idea how to hook it up in my application.

**Comment by vporton** [comment]
With the UI you can buy Item 1, subscribe to a subscription, unsubscribe, pay for a subscription manually (non-recurringly).

See https://github.com/vporton/django-debits/wiki about the API.

**Comment by antizeus** [comment]
I got that however when i proceed with the actions i get no response on the UI at all. I see no changes in my actual paypal testing account. It seems it is broken. How do you set it up in your settings.py at all?

**Comment by vporton** [comment]
Apparently you expect to test it from command line. No, after starting `runserver`, you need to open it in your web browser. The rest UI is quite easy.

**Comment by vporton** [comment]
See the wiki about `settings.py`.

**Comment by antizeus** [comment]
I told you that i expect to test it from the browser but nothing happens when i just press the buttons. It doesn't happen anything with my paypal testing account it doesn't lead me to any page assuring the purchase was done.

**Comment by antizeus** [comment]
It seems everything is set up. Your code seems to be broken. Are you sure you tested it correctly? How can you charge money on it?

**Comment by vporton** [comment]
What exactly happens? Show JavaScript console logs (F12 key) and also possibly screenshots.

You need to setup a PayPal account or PayPal sandbox account. You need to edit `settings.py` (or better create `local_settings.py` in the same directory) and change the paypal account ID and client ID and secret key.

My software does work.

**Comment by vporton** [comment]
Oh, you cannot be lead to a page assuring a purchase was done. PayPal informs Debits that a purchase was done by sending an IPN what happens asynchronously (read: after a minute or so) and you don't get any message for it, except notification emails.

You need to run it on a real server with outside Internet access to receive IPNs.

I hope a future version of my software will work more synchronously (in some modes).

**Comment by antizeus** [comment]
It seems everything is set up but i don't get my paypal account to have the status changed at all. I also looked at the source code of the test and it seems really messy. I can't really understand the demo. I'm sorry but i will not proceed at all with django debits. I'm also going to report this repository to github it seems as a scam.

**Comment by antizeus** [comment]
You have poor engineering skills or you plan to just steal money from people with this package.

**Comment by vporton** [comment]
I disagree that I have poor engineering skills. In fact, I am a very good software engineer. My code is really feature-rich, clean, abstract and modular. Well, it is such at the price of being complex and somehow hard to use. One need to pay the price. It seems you mess messy and complex code.

You have the right to report anything. But it is not like a scam in _any_ way. It is used in a real business project of me running now.

The status may not change for example for the following reasons:

1. You messed real and testing ("sandbox") mode.
1. You entered wrong paypal info.
1. You need to wait more, paypal is sometimes **very** slow.
1. PayPal sandbox is known to have bugs.

**Comment by antizeus** [comment]
You make complex things even more complex. It is pointless. I think you want to scam people. Your code is complex unecessarily you want to steal people. You probably added vulns in it.

**Comment by antizeus** [comment]
Okay, i got it. it is really amazing. Really nice job. You are a really good programmer. Thank you.

**Comment by antizeus** [comment]
Lol, i was being ironic ok?
